File: src/main/java/maven/example/com/receiving/system/IP.java
package maven.example.com.receiving.system;

import maven.example.com.tools.data.Data;
import maven.example.com.tools.data.TypeData;
import maven.example.com.receiving.tools.Receipt;

import java.net.InetAddress;
import java.net.UnknownHostException;

public class IP extends Receipt {
    private static final TypeData TYPE = TypeData.SYSTEM_IP;

    public IP() {
        super(TYPE);
    }

    @Override
    public Data getData() {
        try {
            return new Data(TYPE, InetAddress.getLocalHost().toString());
        } catch (UnknownHostException e) {
            System.err.println("Ошибка при получение IP адреса");
            e.printStackTrace();
            return new Data(TYPE, "empty");
        }
    }
}

File: src/main/java/maven/example/com/receiving/system/IP.kt
package maven.example.com.receiving.system

import maven.example.com.receiving.utility.Receipt
import maven.example.com.utility.data.Data
import maven.example.com.utility.data.TypeData
import java.net.InetAddress
import java.net.UnknownHostException

class IP : Receipt(TYPE) {

    override val data: Data?
    get() = try {
        Data(TYPE, InetAddress.getLocalHost().toString())
    } catch (e: UnknownHostException) {
        System.err.println("Ошибка при получение IP адреса")
        e.printStackTrace()
        Data(TYPE, "empty")
    }

    companion object {
        private val TYPE = TypeData.SYSTEM_IP
    }
}
